### PR TITLE
Fix willReceiveProps

### PR DESCRIPTION
**Pull request by guljeny**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,5 +3,6 @@
 /dist/
 /docs/
 /lib/
+/.idea/
```
Comment (JelF) [inline .gitignore]: this should be in global gitignore

File: src/components/Counter.js [hunk 2]
```diff
@@ -147,9 +161,10 @@ export default class Counter extends React.Component {
       return this.stop()
     }
 
-    const timestamp = this.props.direction === 'down'
-      ? newTimeDiff
-      : this.state.initialTimeDiff - newTimeDiff
+    const timestamp =
+      this.props.direction === 'down'
```
Comment (JelF) [inline src/components/Counter.js]: Does code style accept multiline ternaries outside jsx?

**Review by JelF — CHANGES_REQUESTED** [review]

File: src/components/Counter.js [hunk 3]
```diff
@@ -258,10 +274,6 @@ export default class Counter extends React.Component {
         </div>
       )
     })
-    return (
-      <div className={this.getCSSClasses()}>
-        {segments}
-      </div>
-    )
+    return <div className={this.getCSSClasses()}>{segments}</div>
```
Comment (JelF) [inline src/components/Counter.js]: brackets are requried in `return(<jsx ... />)`

**Review by JelF — DISMISSED** [review]

File: src/components/Counter.js (hunk 3 above)
Comment (akxcv) [reply src/components/Counter.js]: No, brackets are only required if the JSX is multiline.

**Comment by akxcv** [comment]
@JelF the Circle CI build is failing because of some changes on Sauce Labs' part, it's not a fault of this PR. Please review the changes once again.

File: src/components/Counter.js [hunk 4]
```diff
@@ -112,6 +112,20 @@ export default class Counter extends React.Component {
         this.start()
       }
     }
+    if (
```
Comment (JelF) [inline src/components/Counter.js]: maybe guard clause here?

**Review by JelF — COMMENTED** [review]
I generally don't want accept this request beacause it requires serious testing and i have no time to do it.

Counter has 16 props and i am even not sure that change of any other prop (i.e. seconds) would not cause same bug. 

I also think this logic should be tested in feature test instead of unit.

However i have no reason to reject it's merging if someone else would check it

**Comment by akxcv** [comment]
@JelF the feature tests are green. Besides, this is a standard change for a React component and I see no reason not to merge this. AFAIK, this won't break anything.

**Pull request merged**